contracts/src/examples/modules/SchemaCheckerModule.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import { AbstractModule } from "../../interface/AbstractModule.sol";
+import { AttestationPayload } from "../../types/Structs.sol";
+
+/**
+ * @title Schema checker Module
+ * @notice This contract illustrates a valid Module that is used to check the schema that an attestation is based on
+ */
+contract SchemaCheckerModule is AbstractModule {
+  error InvalidSchemaId();
+  struct Profile {
```

Comment:
I don't understand why there is this struct and we are decoding the attestationData

Reply:
Just trying to demonstrate how to encode & decode complex struct in the attestation data

Reply:
Maybe we just need to add a comment saying just that: this is an example of how to check the attestation payload encoding/decoding, based on a "Profile Schema"?

Reply:
Comments had been added @orbmis